common/static/common/js/discussion/views/discussion_thread_list_view.js

```diff
@@ -535,24 +519,39 @@
                                 self.addSearchAlert(gettext('No posts matched your query.'));
                             }
                             self.displayedCollection.reset(self.collection.models);
-                            if (text) {
-                                return self.searchForUser(text);
-                            }
                         }
                         return response;
                     }
                 });
             };
 
-            DiscussionThreadListView.prototype.searchForUser = function(text) {
+            DiscussionThreadListView.prototype.searchForUser = function(text, $searchInput) {
                 var self = this;
+                this.clearSearchAlerts();
+                this.clearFilters();
                 return DiscussionUtil.safeAjax({
                     data: {
                         username: text
                     },
                     url: DiscussionUtil.urlFor('users'),
                     type: 'GET',
                     dataType: 'json',
+                    $loading: $,
```

Comment:
I have replaced the method, so when you search something it shows the loading image. Firstly we called the `searchFor` but now we calling `searchForUser `.